packages/turf-boolean/touch.js

```diff
@@ -0,0 +1,98 @@
+var inside = require('@turf/inside');
+var lineOverlap = require('@turf/line-overlap');
+var polyToLinestring = require('@turf/polygon-to-linestring');
+
+var internalHelpers = require('./helpers');
+var checkIfCoordArraysMatch = internalHelpers.checkIfCoordArraysMatch;
+var isPointOnLineSegment = internalHelpers.isPointOnLineSegment;
+
+module.exports = function (feature1, feature2) {
+    if (feature1.geometry.type === 'Point' && feature2.geometry.type === 'LineString') {
+        return isPointAtEndOfLine(feature1, feature2);
+    }
+    if (feature1.geometry.type === 'MultiPoint' && feature2.geometry.type === 'LineString') {
+        return isMultiPointAtEndOfLine(feature1, feature2);
+    }
+    if (feature1.geometry.type === 'Point' && feature2.geometry.type === 'Polygon') {
+        return isPointOnPolygon(feature1, feature2);
+    }
+    if (feature1.geometry.type === 'MultiPoint' && feature2.geometry.type === 'Polygon') {
+        return isMultiPointOnPolygon(feature1, feature2);
+    }
+    // if (feature1.geometry.type === 'LineString' && feature2.geometry.type === 'LineString') {
+    //     return isLineStringOnLineString(feature1, feature2);
+    // }
+    // if (feature1.geometry.type === 'LineString' && feature2.geometry.type === 'Polygon') {
+    //     return isLineStringOnPolygon(feature1, feature2);
+    // }
+};
+
+function isPointAtEndOfLine(Point, LineString) {
+    if (checkIfCoordArraysMatch(Point.geometry.coordinates, LineString.geometry.coordinates[0])) {
+        return true;
+    }
+    if (checkIfCoordArraysMatch(Point.geometry.coordinates, LineString.geometry.coordinates[LineString.geometry.coordinates.length - 1])) {
+        return true;
+    }
+    return false;
+}
+
+function isMultiPointAtEndOfLine(MultiPoint, LineString) {
+    var i;
+    var onStartOrEnd = 0;
+    var pointIsSomewhereOnLine = 0;
+    var lastLinestringCoordIndex = LineString.geometry.coordinates.length - 1;
+    for (i = 0; i < MultiPoint.geometry.coordinates.length; i++) {
+        if (checkIfCoordArraysMatch(MultiPoint.geometry.coordinates[i], LineString.geometry.coordinates[0])) {
+            onStartOrEnd++;
+            continue;
+        }
+        if (checkIfCoordArraysMatch(MultiPoint.geometry.coordinates[i], lastLinestringCoordIndex)) {
+            onStartOrEnd++;
+            continue;
+        }
+        for (var lineI = 0; lineI < LineString.geometry.coordinates.length - 1; lineI++) {
+            if (isPointOnLineSegment(LineString.geometry.coordinates[lineI], LineString.geometry.coordinates[lineI + 1], MultiPoint.geometry.coordinates[i])) {
+                pointIsSomewhereOnLine++;
+            }
+        }
+    }
+    return onStartOrEnd === 1 && pointIsSomewhereOnLine === 0;
```

Comment:
Just a nit-pick. You can probably return inside the loop if the second criteria is broken. That way you don't iterate through all the points if you don't need to. Also, I think the first criteria should read `onStartOrEnd > 0`. It could be both ends that are touched.